### PR TITLE
gen: extend +vats and |suspend to take lists of desks

### LINKED ISSUE #6470: gen: +vats doesn't handle the %kids presentation well
From [this comment](https://github.com/urbit/urbit/pull/6395#issuecomment-1504437791) on #6395:

`+vats` doesn't handles the `%kids` presentation well:
```
+vats
<snip>
::
 %kids %kids %cz hash:     0v14.4jub3.fe4i1.5pfa6.do69s.fqkh7.tf6o9.g7b68.pgdeu.jjec8.u8517::
```
```
+vat %kids
%kids %cz hash:     0v14.4jub3.fe4i1.5pfa6.do69s.fqkh7.tf6o9.g7b68.pgdeu.jjec8.u8517
```

### PR DESCRIPTION
**Pull request by silnem2**
Extends +vats to take an optional list of one or more desks; if +vats is run with a list of desks and a =filt argument the output will contain only the members of the list that match the filter. Also removes the now-redundant +vat generator, updates |suspend to take a list of desks, and refactors +vats and related functions in /sur/hood. Resolves #6470 and #6477.

### CODE REVIEW

File: pkg/arvo/gen/vat.hoon
Comment (jalehman) [inline pkg/arvo/gen/vat.hoon:1]: Instead of removing +vat, I'd rather it output a deprecation notice and be refactored in terms of the newly extended `+vats`. That'll prevent breakage to anyone's workflow and teach them how to use the new capabilities in one fell swoop.

**Review by jalehman — COMMENTED** [review]
This resolves my previous, although I'll defer to @belisarius222 on an actual code review.

**Review by belisarius222 — APPROVED** [review]
I tested this on a fakezod, and it all works right for me.

The non-verbose printing is still wonky, but that behavior is unchanged from the previous version of +vats, so it's not a regression.

**Review by belisarius222 — CHANGES_REQUESTED** [review]
On second thought, the changes to `|suspend`, while I don't think they're risky, do imply a backward-incompatible change to the Kiln API, which someone might be depending on.  There are three options here:
1) wait until we have confirmation that all "systemically important" agents that are clients of Kiln's `%suspend` API are manually notified so they can change their use of this before release
2) split this into two PRs and merge the `|suspend` changes into a later Kelvin branch
3) revert the changes to Kiln's `%kiln-suspend` poke datatype, and instead change the `|suspend` generator to use a new `%kiln-suspend-many` mark, handled in a new `+poke-suspend-many` arm in Kiln.  `+poke-suspend` can be rewritten to call into that arm with a single-element list of desks to avoid code duplication.

Option 1 isn't great.  Options 2 and 3 are both acceptable.  I would prefer option 3, to reduce compatibility work for client apps.

**Pull request merged**
